src/components/InputField/InputField.tsx

```diff
@@ -183,6 +207,14 @@ export const InputField: InputFieldType = forwardRef(
       ? ariaDescribedBy || generatedAriaDescribedById
       : undefined;
 
+    const fieldLengthCountClassName = clsx(
+      (textExceedsMaxLength || textExceedsRecommendedLength) &&
+        styles['input-field--invalid-length'],
+    );
+
+    // Pick the smallest of the lengths to set as the maximum value allowed
+    const maxLengthShown = getMinValue(maxLength, recommendedMaxLength);
```

Comment:
If `maxLength` and `recommendedMaxLength` are numbers (e.g. if they have default values or fallbacks), may be able to use `Math.min`?